src/rpc/mining.cpp

```diff
@@ -578,24 +585,36 @@ UniValue getblocktemplate(const UniValue& params, bool fHelp)
             nTransactionsUpdatedLastLP = nTransactionsUpdatedLast;
         }
 
-        // Release the wallet and main lock while waiting
-        LEAVE_CRITICAL_SECTION(cs_main);
         {
-            checktxtime = boost::get_system_time() + boost::posix_time::minutes(1);
+            checktxtime = boost::get_system_time() + boost::posix_time::seconds(10);
 
+            int nHeight = chainActive.Tip()->nHeight;
             boost::unique_lock<boost::mutex> lock(csBestBlock);
             while (chainActive.Tip()->GetBlockHash() == hashWatchedChain && IsRPCRunning())
             {
-                if (!cvBlockChange.timed_wait(lock, checktxtime))
-                {
-                    // Timeout: Check transactions for update
-                    if (mempool.GetTransactionsUpdated() != nTransactionsUpdatedLastLP)
-                        break;
-                    checktxtime += boost::posix_time::seconds(10);
+                // Before waiting, generate the coinbase for the block following the next
+                // block (since this is cpu-intensive), so that when next block arrives,
+                // we can quickly respond with the template for following block.
```

Comment:
Add a note here that the time spent creating the coinbase is not before/in addition to the wait time, but within it (due to `cvBlockChange.timed_wait` taking an absolute time, not a duration).